Week1 [STEP 3] Hisop

Hi-sop opened the pull request:
@juun97 
STEP3 PR

## 고민했던 점
- 길이와 막대 길이는 Int로 큰 값이 들어와도 반복문으로 계속 출력하면 되어 예외처리는 없어도 괜찮다고 생각했습니다. (음수는 고려하지 않았습니다)
- width가 제시되지 않은 점, 예시에서 모두 길이가 고정되어 있었으므로 몸통과 토핑은 고정된 공간을 가지고 있다고 가정했습니다. 따라서 prefix를 사용해 각각 3, 1 길이의 String을 넘겨줄 수 있도록 했습니다.
 (body의 값이 3이하일 경우는 고려하지 않았습니다)
- prefix는 subString타입을 반환하기 때문에 String으로 변환했습니다.
- 매개변수 기본값을 " "로 두어 topping을 전달받지 못하더라도 한칸 띄워줄 수 있도록 했습니다.

## 조언을 얻고 싶은 부분
- 이번 플로우차트는 window 환경에서 작성해 remote에서 바로 Add file했습니다. 이렇게 PR을 해도 상관 없을까요?
- 커밋 컨벤션은 [Git-Commit-Message-Convention](https://velog.io/@archivvonjang/Git-Commit-Message-Convention) 해당 링크를 참조하여 작성했는데, 링크 페이지의 서식대로 계속 작성해도 좋을지 궁금합니다.

Commit:
Feat: 빼빼로만들기 구현
과제의 요구사항에 따라 함수 분리
긴 길이의 String이 들어왔을때 prefix로 예외처리

Hi-sop committed:
Feat: 플로우차트 추가

받은 데이터를 그대로 출력하기에 플로우차트가 짧았다.

juun97 reviewed (commented):
## 고민했던 점
> - 길이와 막대 길이는 Int로 큰 값이 들어와도 반복문으로 계속 출력하면 되어 예외처리는 없어도 괜찮다고 생각했습니다. (음수는 고려하지 않았습니다)
>- width가 제시되지 않은 점, 예시에서 모두 길이가 고정되어 있었으므로 몸통과 토핑은 고정된 공간을 가지고 있다고 가정했습니다. 따라서 prefix를 사용해 각각 3, 1 길이의 String을 넘겨줄 수 있도록 했습니다.
 (body의 값이 3이하일 경우는 고려하지 않았습니다)
>- prefix는 subString타입을 반환하기 때문에 String으로 변환했습니다.
>- 매개변수 기본값을 " "로 두어 topping을 전달받지 못하더라도 한칸 띄워줄 수 있도록 했습니다.

코멘트에서 얘기 드리겠습니다 👍🏻

## 조언을 얻고 싶은 부분
> - 이번 플로우차트는 window 환경에서 작성해 remote에서 바로 Add file했습니다. 이렇게 PR을 해도 상관 없을까요?

네. github 의 레포지토리는 로컬 컴퓨터의 OS 와 관련없이 관리되는 Remote Repository 이므로 문제 없습니다.


>- 커밋 컨벤션은 [Git-Commit-Message-Convention](https://velog.io/@archivvonjang/Git-Commit-Message-Convention) 해당 링크를 참조하여 작성했는데, 링크 페이지의 서식대로 계속 작성해도 좋을지 궁금합니다.

네. 실제로도 해당 컨벤션은 저도 자주 사용하는 컨벤션입니다. 
깃 협업에서 커밋 컨벤션을 맞추는 것은 서로 소통하는데 있어서 매우 중요한 부분이고, 지금 부터 고민 하신점 아주아주 칭찬 드립니다 👍🏻

Commit:
Fix: 피드백에 따른 일부 코드 수정
printOutInfo의 매개변수명 수정 height -> bodyHeight
몸통과 토핑을 매개변수로 받아 그리는 함수, 실제로 몸통을 출력하는 함수 다시 구분함
height 음수값 들어올경우 return코드 추가

Commit:
Style: 정보 출력 String 각각 3, 1로 제한
인자 전달시 prefix사용하도록 변경
